### PR TITLE
Add question answers word cloud feature

### LINKED ISSUE #772: [Live-Site] Word Cloud Question/Answer Feature
### Issue Description

We have to create realtime word cloud feature in which host can ask question in realtime to connected clients(guest, mavens, moderators) and the answers will form word cloud in realtime as answers comes in.

### Expected Behavior

n/a

### Current Behavior

n/a

### Screenshots

[If applicable, add screenshots or images that help illustrate the issue. You can use the following Markdown syntax to embed images:]

### Reproducibility

 - [ ] This issue is reproducible
-  [ ] This issue is not reproducible

### Steps to Reproduce
1. [Step 1]
2. [Step 2]

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [x] Medium
 - [ ] Low

### Additional Information

[Include any additional context, logs, or details that might be helpful in addressing the issue.]

### Checklist

- [ ] I have read and followed the project's code of conduct.
- [ ] I have searched for similar issues before creating this one.
- [ ] I have provided all the necessary information to understand and reproduce the issue.
- [ ] I am willing to contribute to the resolution of this issue.

---

Thank you for contributing to our project! We appreciate your help in improving it.

### PR DESCRIPTION
**Pull request by satyam73**
### What is the change?

In this PR I have added word cloud question and answer feature.
Sequence diagram - https://lucid.app/lucidspark/02e146bd-6d0f-4d4e-a425-f811c9289e49/edit?viewport_loc=-5030%2C-192%2C3515%2C2201%2C0_0&invitationId=inv_4a2f8f42-57df-473f-86a1-caea4c57d619
Backend Ticket - https://github.com/Real-Dev-Squad/website-backend/issues/1805

### Is Development Tested?

- [x] Yes
- [ ] No

### Before :
n/a

### After :

https://github.com/Real-Dev-Squad/website-www/assets/85296770/b7321089-ea8e-4ebd-8df9-2a0a2085f8a2

**Note :**
- Tests will be handled in another PR since this one is getting big, the PR for tests is attached below.
- https://github.com/Real-Dev-Squad/website-www/pull/805

### CODE REVIEW

**Comment by satyam73** [comment]
Q. Why eslint rule has self-assign equals to warn?
Ans. So for triggering an update in the ember @tracked object we need to self assign and eslint has property which was giving error so set it to warn.
More in docs - https://guides.emberjs.com/release/upgrading/current-edition/tracked-properties/#toc_plain-old-javascript-objects-pojos
image
> Image attached for reference

File: app/templates/live.hbs [hunk 1]
```diff
@@ -1,14 +1,25 @@
-{{page-title 'Live'}}
+{{page-title "Live"}}
 
-<div id='live' data-test-live>
+<div id="live" data-test-live>
   {{#if this.liveService.isJoined}}
+    {{!-- {{#if true}} --}}
```
Comment (ajoykumardas12) [inline app/templates/live.hbs:5]: Unnecessary comment?

File: app/templates/live.hbs [hunk 2]
```diff
@@ -47,34 +58,35 @@
...
     {{#if this.liveService.isLoading}}
-      <div data-test-loading class='live__loader-container'>
-        <Fa-Icon @size='2x' @icon='circle-notch' @spin={{true}} />
+      <div data-test-loading class="live__loader-container">
+        <Fa-Icon @size="2x" @icon="circle-notch" @spin={{true}} />
       </div>
     {{else}}
       {{#if this.liveService.isJoined}}
-        <div class='live__container'>
+        {{!-- {{#if true}} --}}
```
Comment (ajoykumardas12) [inline app/templates/live.hbs:75]: Unnecessary comment?

File: app/templates/live.hbs [hunk 3]
```diff
@@ -47,34 +58,35 @@
...
+        <div class="live__container">
           <div
-            class='live__video-container
-              {{if this.liveService.isScreenShareOn "bg-black" ""}}'
+            class="live__video-container
+              {{if this.liveService.isScreenShareOn 'bg-black' ''}}"
             data-test-video-container
           >
             {{! TODO - add skeleton to the whole structure }}
```
Comment (vinit717) [inline app/templates/live.hbs:82]: Is this still a todo?

File: app/styles/ask-question-modal.module.css [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+.ask-question-modal__checkbox-container {
+  display: flex;
+  gap: 0.5rem;
+}
+
+.ask-question-modal__checkbox {
+  appearance: none;
+  background-color: #fff;
```
Comment (vinit717) [inline app/styles/ask-question-modal.module.css:55]: I think we have css variable for white can you use that

File: app/styles/answer-view-card.module.css [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  cursor: pointer;
+}
+
+.answer-view-card--pending {
+  background-color: #ffffe0;
+}
+
+.answer-view-card--approved {
+  background-color: #98fb98;
+}
```
Comment (vinit717) [inline app/styles/answer-view-card.module.css:31]: make a variable for this is not present

File: app/services/survey.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  async answerRejectHandler(id) {
+    const rejectionPayload = {
+      status: ANSWER_STATUS.REJECTED,
+    };
+    try {
+      const rejectResponse = await fetch(`${APPS.API_BACKEND}/answers/${id}`, {
+        ...PATCH_API_CONFIGS,
+        body: JSON.stringify(rejectionPayload),
```
Comment (vinit717) [inline app/services/survey.js:109]: Why we are not using ember data?

File: app/components/ask-question-modal.hbs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      <label
+        class="ask-question-modal__checkbox-label"
+        for="is-max-characters-checked"
+      >Do you want answer to be of limited characters?</label>
+    </div>
+    <input
+      type="number"
+      name="max-characters"
+      id="max-characters"
+      min="1"
```
[the remark this answers is not in the document]
Comment (satyam73) [reply app/components/ask-question-modal.hbs:34]: nope this constant is for different thing

File: app/services/survey.js [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+      if (!approveResponse.ok) throw new Error();
+
+      this.toast.success(
+        'Answer approved successfully',
+        'Success',
+        TOAST_OPTIONS,
+      );
+    } catch (error) {
+      console.error('Error while approving answer: ', error);
```
[the remark this answers is not in the document]
Comment (satyam73) [reply app/services/survey.js:97]: We have it from start that's why added this to get the client know that some error happened

File: app/templates/live.hbs (hunk 3 above)
Comment (satyam73) [reply app/templates/live.hbs:82]: yes

File: app/services/survey.js (hunk 6 above)
Comment (satyam73) [reply app/services/survey.js:109]: The apis were slight different so haven't used ember data

File: app/components/answer-view-card.hbs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      {{! template-lint-disable no-duplicate-id}}
+      @id={{@id}}
+      @class="icon-button--sm icon-button--red-outlined"
+      @title="Reject Answer"
+      @onClick={{@onReject}}
+      @icon="material-symbols:close-small-outline-rounded"
+    />
+    <Reusables::IconButton
+      {{!  TODO- fix this unique id eslint problem }}
+      {{! template-lint-disable no-duplicate-id}}
```
[the remark this answers is not in the document]
Comment (satyam73) [reply app/components/answer-view-card.hbs]: fixed it in this one👍

File: app/services/survey.js (hunk 6 above)
Comment (vinit717) [reply app/services/survey.js]: Can you tell  why we avoid ember-data in this case?

Comment (satyam73) [reply app/services/survey.js]: Because all things are happening in realtime we don't need functionality like caching that's why.

**Review by ajoykumardas12 — APPROVED** [review]

**Pull request merged**